packages/fxa-auth-server/scripts/delete-unverified-accounts/delete-unverified-accounts.ts

```diff
@@ -0,0 +1,308 @@
+    program.startDate && program.endDate && program.endDate > program.startDate;
+  const reason = ReasonForDeletion.Unverified;
+  const taskLimit = program.taskEnqueueLimit
+    ? parseInt(program.taskEnqueueLimit)
+    : 200;
+
+  if (!hasUid && !hasEmail && !hasDateRange) {
+    throw new Error(
+      'The program needs at least a uid, an email, or valid date range.'
+    );
+  }
+  if ((hasUid || hasEmail) && hasDateRange) {
+    throw new Error(
+      'Sorry, but the script does not support uid/email arguments and a date range in the same invocation.'
+    );
+  }
+  if (limit <= 0) {
+    throw new Error('The limit should be a positive integer.');
+  }
+
+  const useSpecifiedAccounts = hasUid || hasEmail;
+  const useDateRange = !useSpecifiedAccounts && hasDateRange;
+
+  if (isDryRun) {
+    console.log(
+      'Dry run mode is on.  It is the default; use --dry-run=false when you are ready.'
+    );
+
+    return dryRun(program, useSpecifiedAccounts, useDateRange, limit);
+  }
+
+  const config = appConfig.getProperties();
+  const log = initLog({
+    ...config.log,
+  });
+  const statsd = new StatsD({ ...config.statsd });
+  const redis = initRedis(
+    { ...config.redis, ...config.redis.sessionTokens },
+    log
+  );
+  const db = DB(
+    config,
+    log,
+    Token(log, config),
+    random.base32(config.signinUnblock.codeLength) as any // TS type inference is failing pretty hard with this
+  );
+  const fxaDb = await db.connect(config, redis);
+
+  Container.set(AppConfig, config);
+  Container.set(AuthLogger, log);
+
+  const authFirestore = setupFirestore(config);
+  Container.set(AuthFirestore, authFirestore);
+  const currencyHelper = new CurrencyHelper(config);
+  Container.set(CurrencyHelper, currencyHelper);
+  const stripeHelper = createStripeHelper(log, config, statsd);
+  Container.set(StripeHelper, stripeHelper);
+
+  const accountTasks = AccountTasksFactory(config, statsd);
+
+  if (useSpecifiedAccounts) {
+    const { uids, emails } = limitSpecifiedAccounts(program, limit);
+
+    for (const x of uids) {
+      const acct = await fxaDb.account(x);
+      if (acct && acct.emailVerified) {
+        console.error(`Account with uid ${x} is verified.  Skipping.`);
+        continue;
+      }
+      const result = await accountTasks.deleteAccount({
+        uid: x,
+        customerId: (await getAccountCustomerByUid(x))?.stripeCustomerId,
+        reason,
+      });
+      console.log(`Created cloud task ${result} for uid ${x}`);
+    }
+
+    for (const x of emails) {
+      const acct = await fxaDb.accountRecord(x);
+      if (acct && acct.emailVerified) {
+        console.error(`Account with email ${x} is verified.  Skipping.`);
+        continue;
+      }
+      const result = await accountTasks.deleteAccount({
+        uid: acct.uid,
+        customerId: (await getAccountCustomerByUid(acct.uid))?.stripeCustomerId,
+        reason,
+      });
+      console.log(`Created cloud task ${result} for uid ${x}`);
```

Comment:
nit: should this say `uid ${x}` or `email ${x}` since we're doing `for (const x of emails) {...}` here?